### PR TITLE
GH-550: master to 2.2; fix tangles

### PR DESCRIPTION
**Pull request by garyrussell**
Fixes https://github.com/spring-projects/spring-kafka/issues/550

- package tangle `....listener` and `....listener.config` - remove config package
- class tangles between `ContainerProperties` and the listener containers
  - AckMode moved to properties
  - Error handler setters moved from properties to containers
- class tangle between listener adapter and handler adapter due to `ResultHolder`

### CODE REVIEW

**Comment by garyrussell** [comment]
__Do not squash commits.__

**Comment by garyrussell** [comment]
I have created the 2.1.x branch.

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java [hunk 1]
```diff
@@ -89,27 +88,27 @@
...
-			}
-			catch (NoSuchMethodException | SecurityException e) {
-				throw new RuntimeException("Failed to determine TestUtils.createBrokerConfig() method");
-			}
-		}
-		else {
+//		clientVersion = AppInfoParser.getVersion();
+//		if (clientVersion.startsWith("1.1.")) {
+//			try {
+//				testUtilsCreateBrokerConfigMethod = TestUtils.class.getDeclaredMethod("createBrokerConfig",
```
[the remark this answers is not in the document]
Comment (garyrussell) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java]: I decided to just comment it out to make it easier to reinstate if we have a similar problem with 1.2.0. I can remove it entirely if you insist.

File: build.gradle [hunk 2]
```diff
@@ -78,7 +78,8 @@ subprojects { subproject ->
 		junitJupiterVersion = '5.1.0'
 		junitPlatformVersion = '1.1.0'
 		junitVintageVersion = '5.1.0'
-		kafkaVersion = '1.0.1'
+		kafkaVersion = '1.1.0'
+		log4jVersion = '2.10.0'
```
[the remark this answers is not in the document]
Comment (garyrussell) [reply build.gradle]: It doesn't come from kafka - they use slf4j now. https://kafka.apache.org/documentation/#upgrade_110_notable

[the remark this answers is not in the document]
Comment (garyrussell) [reply build.gradle]: Yes; I switched to log4j2.

File: build.gradle [hunk 3]
```diff
@@ -184,7 +188,6 @@ project ('spring-kafka') {
 		compile ("org.apache.kafka:kafka-streams:$kafkaVersion") {
 			optional it
 			exclude group: 'org.slf4j', module: 'slf4j-api'
```
[the remark this answers is not in the document]
Comment (garyrussell) [reply build.gradle]: Given that we need to add the `log4j-slf4j-impl` to the classpath (to bridge from slf4j to log4j2), I think we need the api to be the same version.

[the remark this answers is not in the document]
Comment (garyrussell) [reply build.gradle]: Doh... pushing

**Comment by garyrussell** [comment]
Sorry - forgot about the Windows problem. You should squash the last 3 commits; retaining 2 commit.

Yes, please add the #278 comment (to the squashed second commit).

Thanks.

**Comment by artembilan** [comment]
I also noticed that we have several not closed `KafkaConsumer`s in some tests.
Fixing...
Thanks for confirmation!

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/b048aaa8f00045e4cc65d5d137b1aa372beca3a2 and https://github.com/spring-projects/spring-kafka/commit/9f724ea27a2b403d3dcc846ce5c7c6805cdff108

**Pull request closed without merging**